js/src/table.ts

```diff
@@ -93,15 +96,34 @@ export class Table<T extends { [key: string]: DataType; } = any>
     }
 
     /** @nocollapse */
-    public static fromVectors<T extends { [key: string]: DataType; } = any>(vectors: VType<T[keyof T]>[], names?: (keyof T)[]) {
-        return new Table(RecordBatch.from(vectors, names));
+    public static fromVectors<T extends { [key: string]: DataType; } = any>(vectors: Vector<T[keyof T]>[], fields?: (keyof T | Field<T[keyof T]>)[]) {
+        return Table.new<T>(vectors, fields);
```

Comment:
I like the pattern where `from` methods are intended for conversions from other types and `new` methods are effectively constructors where we can use generics.

Do you think we should just go ahead and get rid of this to follow that pattern? or mark it deprecated or something? It's only been part of 0.4.0 for a couple of days I don't think anyone's using it.

Reply:
I'm down for either. I couldn't remember if this method predated 0.4 or not, so I left it unchanged.

Reply:
I think anyone would expect constructors instead of `.new` methods. Since we are pre 1.0 everything is fair game and you can be backwards incompatible in minor version bumps.

Reply:
@domoritz agreed, I say let's just rename this (and `RecordBatch.from`) as an overload of `new`, we haven't even put out a blog post or anything touting these functions in 0.4.0. We can just tout the `new` versions instead.